src/controller/posts/updatePost.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable camelcase */
+const updatePost = require('../../database/queries/post/updatePost');
+const updateSchema = require('../../utils/validation/posts/updatePost.validation');
+const getPostById = require('../../database/queries/post/getPostById');
+const CustomError = require('../../utils/helpers/customError');
+
+const updatePosts = (req, res, next) => {
+  const { id: postID } = req.params;
+  getPostById(postID)
+    .then((post) => {
+      if (!post.rowCount) {
+        throw new CustomError(401, 'Post not found');
+      }
+      return post.rows[0];
+    })
+    .then((post) => {
+      if (post.user_id !== req.user.id) {
+        throw new CustomError(401, 'unauthorized');
+      }
+    })
+    .then(() => updateSchema.validateAsync(req.body))
+    .then((validated) => {
+      if (validated.images === '') {
+        const { title, content } = validated;
+        return updatePost({ title, content }, postID);
+      }
+      return updatePost(validated, postID);
+    })
+    .then((updatedPost) => {
+      const { username } = req.user;
+      const {
+        id,
+        title,
+        content,
+        images,
+        user_id,
+        created_at,
+        upvotes,
+        downvotes,
+        comment_count,
+      } = updatedPost.rows[0];
+      return res.json({
+        error: false,
+        data: {
+          id,
+          title,
+          content,
+          images,
+          user_id,
+          created_at,
+          username,
+          upvotes,
+          downvotes,
+          comment_count,
+        },
+      });
```

Comment:
Instead of manually assigning all the values to a new object, you can simply use the spread operator to include all the properties of the returned row object and add the username property to it. 
Here's an example of how the code can be simplified:
```js
const { username } = req.user;
return res.json({
  error: false,
  data: {
    ...updatedPost.rows[0],
    username,
  },
});
```
And if you do not want to send all the data returned from this query, you can return all columns that you want only and ignore any columns that you do not want to appear in the response.